src/project-css/app/Helpers/CMSHelper.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * @author Ajay Krishna Teja Kavur
+ * @author Tracy A McCormick <tam0013@mail.wvu.edu>
+ */
+
+namespace App\Helpers;
+
+use App\Models\CMSRecords;
+use App\Models\Collection;
+use App\Helpers\CSVHelper;
+use App\Helpers\TableHelper;
+
+class CMSHelper {
+    /**
+     * CMS Helper
+     *
+     * These are various functions that help with processing
+     * cms files.
+     *
+     */
+
+
+     /**
+      * Check the Record Types table for a matching header for the table
+      *
+      * @param integer $collctnID Contains the collection ID
+      * @param string $recordType Contains the Record type 1A, 1B, etc.
+      * @param integer $fieldCount Contains number of fields expected for the Record
+      *
+      * @author Tracy A. McCormick <tam0013@mail.wvu.edu>
+      * @return unserialized Array if found or null
+      */
+     public function getCMSFields($collctnId, $recordType, $fieldCount) {
+       if (CMSRecords::isCMSRecord($recordType)) {
+
+        // get collection record so we can set the cms ID
+        $thisClctn = Collection::findorFail($collctnId);
+
+        $response = CMSRecords::findCMSHeader($recordType, $fieldCount);
+
+        // if only one item is found we return that result
+        if (count($response) == 1) {
+          // set cmsID in Collection if not already set
+          if ($thisClctn->cmsId == null) {
+            $thisClctn->setCMSId($collctnId, $response[0]->cmsId);
+          }
+
+          // first returned response should be the closest match
+          return unserialize($response[0]->fieldNames);
+        }
+        elseif (count($response) > 1) {
+          for ($arrayPos = 0; $arrayPos < count($response); $arrayPos++) {
+            if ($response[$arrayPos]->cmsId === $thisClctn->cmsId) {
+              return unserialize($response[$arrayPos]->fieldNames);
+            }
+          }
+        }
+       }
```

Comment:
Looking at the logical structure here there is a huge branch condition assignment.  

if +1 
   nested if +2 
elseif  +1
   loop n+
       nested if n+2

assuming the loop contains only 1 item, you pass the branch conditional assignments, otherwise you fail. The branch conditional assignment keeps the single responsibility pattern a top thought of your application.